Content.IntegrationTests/Tests/GameObjects/Components/Mobs/ActionsComponentTests.cs

```diff
@@ -182,5 +187,159 @@ public async Task GrantsAndRevokesActionsTest()
             });
         }
 
+        [Test]
+        public async Task GrantsAndRevokesItemActions()
+        {
+            var (client, server) = await StartConnectedServerClientPair();
+
+            await server.WaitIdleAsync();
+            await client.WaitIdleAsync();
+
+            var serverPlayerManager = server.ResolveDependency<Robust.Server.Interfaces.Player.IPlayerManager>();
+            var serverEntManager = server.ResolveDependency<IEntityManager>();
+            var serverGameTiming = server.ResolveDependency<IGameTiming>();
+
+            var cooldown = Cooldowns.SecondsFromNow(30, serverGameTiming);
+
+            ServerActionsComponent serverActionsComponent = null;
+            ClientActionsComponent clientActionsComponent = null;
+            IEntity serverPlayerEnt = null;
+            IEntity serverFlashlight = null;
+
+            await server.WaitAssertion(() =>
+            {
+                serverPlayerEnt = serverPlayerManager.GetAllPlayers().Single().AttachedEntity;
+                serverActionsComponent = serverPlayerEnt.GetComponent<ServerActionsComponent>();
+
+                // spawn and give them an item that has actions
+                serverFlashlight = serverEntManager.SpawnEntity("FlashlightLantern",
```

Comment:
It is preferable for tests to not rely on external prototypes (#2553). Instead, you can load in prototypes from a string. For example, like `TransformTest`, linked in the issue.

Reply:
I'm not sure how that approach would be done here. This is a content integration test, so OneTimeSetup won't work since that's not running in the actual server / client instance.

If I instead try to load it within the instances, like so:

```csharp
private void LoadPrototype()
{
    var manager = IoCManager.Resolve<IPrototypeManager>();
    manager.LoadFromStream(new StringReader(PROTOTYPES));
    manager.Resync();
}
        
[Test]
public async Task GrantsAndRevokesItemActions()
{
    var (client, server) = await StartConnectedServerClientPair();

    await client.WaitAssertion(LoadPrototype);
    await server.WaitAssertion(LoadPrototype);
    
    await server.WaitIdleAsync();
    await client.WaitIdleAsync();

  // rest of the test
}
```

I get this exception
```
System.InvalidOperationException : Mapped strings are locked, will not add.
   at Robust.Shared.Serialization.RobustMappedStringSerializer.MappedStringDict.AddStrings(YamlStream yaml) in C:\ss14\space-station-14\RobustToolbox\Robust.Shared\Serialization\RobustMappedStringSerializer.MappedStringDict.cs:line 326
   at Robust.Shared.Serialization.RobustMappedStringSerializer.AddStrings(YamlStream yaml) in C:\ss14\space-station-14\RobustToolbox\Robust.Shared\Serialization\RobustMappedStringSerializer.cs:line 530
```

Reply:
I got the test to work with the following:
`            var serverOptions = new ServerIntegrationOptions { ExtraPrototypes = PROTOTYPES };
            var clientOptions = new ClientIntegrationOptions { ExtraPrototypes = PROTOTYPES };

            var (client, server) = await StartConnectedServerClientPair(serverOptions: serverOptions, clientOptions: clientOptions);`

Reply:
ah I missed that capability. fixed it.